ethereum/circuits/lib/src/merkle_patricia_proofs/README.md

```diff
@@ -60,3 +66,7 @@ struct Proof<MAX_DEPTH_NO_LEAF, MAX_LEAF_LEN> {
 **Why is the maximum length of a node equal to 532?**
 A node is an RLP-encoded branch or extension node which means that apart from the data it holds, it additionally consists of RLP headers. If a node is a branch node, in a maximum case, it consists of 16 32-bit hashes and an empty string. When RLP headers are taken into account, the maximum length of a branch node is `3 + (16 * (1 + 32) + 1) = 532`.
 If the node is an extension node, in a maximum case, it consists of a prefixed key part and a hash. As the maximum length of the key this library supports is 495, when accounting for RLP headers, the maximum length of an extension node is `1 + (2 + (1 + 495)) + (1 + 32) = 532`.
+
+## Acknowledgements
+
+This lib was inspired by [noir-trie-proofs](https://github.com/aragonzkresearch/noir-trie-proofs).
```

Comment:
I would add list of changes/improvements, e.g.: using Fragments, BoundedVecs, arbitrary length of leafs (vs internal nodes), etc

Reply:
I propose to use the word rewrite